### DISCUSSION TITLE
Documentation on reverse proxy.

### DISCUSSION THREAD

**Discussion by photounit** [chuot/rdio-scanner#221] [General]
Getting Rdio-Scanner up and running is actually pretty straightforward. The hardest part seems to be getting connected "safely" and reliably to the internet. Would it be possible to have documentation or a tutorial that provides the information on how to configure a reverse proxy or CDN so RS's connection to the web can be done correctly?

**Comment by chuot** [comment]
Although it is not necessary to put a reverse proxy in front of Rdio Scanner to make it more secure, it may be necessary depending on your specific use case.

Depending on the software you want to use for the reverse proxy, there are already many tutorials on this. Here are a few :

- [Nginx](https://docs.nginx.com/nginx/admin-guide/web-server/reverse-proxy/)
- [Apache](https://httpd.apache.org/docs/2.4/howto/reverse_proxy.html)
- [IIS](https://docs.microsoft.com/en-us/iis/extensions/url-rewrite-module/reverse-proxy-with-url-rewrite-v2-and-application-request-routing)

Keep in mind that the reverse proxy must relay standard HTTP connections but also WebSocket connections to the Rdio Scanner server. The mistake I often see is people forgetting to reverse proxy the WebSockets, resulting in a NO LINK message on the main screen.

**Comment by photounit** [comment]
Hi Christian, I've been trying to get Nginx working for months and that's why I posted this....   I'm sure there are others having the same difficulties.   This is the ini file that I'm trying to use with Rdio-Scanner: I also have the web sockets enabled as I read one of your other posts about this being a necessity.    Does this seem to be correct? Thanks for the response.....

db_file = rdio-scanner.db
db_type = sqlite
listen = 192.168.50.126:3000

Apache directives:

RewriteEngine On

RewriteCond %{HTTPS} off 
RewriteCond %{REQUEST_URI} !^/\.well\-known/acme\-challenge/
RewriteRule ^(.*)$ https://%{HTTP_HOST}%{REQUEST_URI} [L,R=301]

RewriteCond %{HTTP:CONNECTION} ^upgrade$ [NC,OR]
RewriteCond %{HTTP:UPGRADE} ^websocket$ [NC]
RewriteRule ^/?(.*)$ ws://192.168.50.126:3000/$1 [P,L]

RewriteCond %{REQUEST_FILENAME} !-d
RewriteCond %{REQUEST_FILENAME} !-f
RewriteRule ^/?(.*)$ http://192.168.50.126:3000/$1 [P,L]

**Reply by noamlivne** [reply]
Regarding "The hardest part seems to be getting connected "safely" and reliably to the internet. ": do you mean that you want to reach Rdio-scanner from the Internet and not only from your localhost?
If so, try Tailscale. It is free and easy to setup. You get a (fictional) IP and you can set it up to reach your home computer from everywhere.

**Comment by chuot** [comment]
I'm a bit confused as you are trying to use Apache HTTP stuff to configure a NGINX server???

I just added some examples for Apache HTTP and NGINX under `/docs/examples`.

If you want to configure a NGINX reverse proxy, then look into `/docs/examples/nginx`. It has been tested and works on Debian 11.

Note that I'm not using NGINX at all, so hopefully someone else here will share his experience with us.
